mkosi/config.py

```diff
@@ -844,8 +857,17 @@ class MkosiConfig:
     def output_dir_or_cwd(self) -> Path:
         return self.output_dir or Path.cwd()
 
-    def workspace_dir_or_cwd(self) -> Path:
-        return self.workspace_dir or Path.cwd()
+    def workspace_dir_or_default(self) -> Path:
+        if self.workspace_dir:
+            return self.workspace_dir
+
+        if os.getenv("XDG_CACHE_HOME") and Path(os.environ["XDG_CACHE_HOME"]).exists():
+            return Path(os.environ["XDG_CACHE_HOME"])
+
+        if os.getenv("HOME") and "/home" in os.environ["HOME"] and (Path(os.environ["HOME"]) / ".cache").exists():
```

Comment:
If mypy doesn't force us to (haven't checked) I think staying with `os.getenv` is a bit more uniform and nicer
```suggestion
        if os.getenv("XDG_CACHE_HOME") and Path(os.getenv("XDG_CACHE_HOME")).exists():
            return Path(os.getenv("XDG_CACHE_HOME"))

        if os.getenv("HOME") and "/home" in os.getenv("HOME") and (Path.home() / ".cache").exists():
```

I'm not sure about the `"/home" in os.getenv("HOME")` check. While generally discouraged, I've seen orgs where home directories where not under `/home` for weird legacy reasons, also, I use mkosi with system users who have their homes under `/var/lib`. I tend to configure cache directories explicitly there, but I'm not sure we should force that.

Reply:
Sounds reasonable enough, but a comment with that reasoning would probably make sense as well as a comment in the man page, that the fallback to `~/.cache` will only be used for non-system users, even if `$HOME` is set for a system user, since this might be a potential footgun in the future with `SetLoginEnvironment=`.